### PR TITLE
Add exclude packages option

### PR DESCRIPTION
**Pull request by Vrixyz**
My guess to support #179 would be to support a new option, to exclude package names.

I will implement that when I have the opportunity to get off my phone 🤞.

### CODE REVIEW

File: src/bloat.ts [hunk 1]
```diff
@@ -99,7 +99,12 @@ export async function runCargoTree(cargoPath: string, packageName: string): Prom
 }
 
 export async function getCargoPackages(cargoPath: string): Promise<Array<CargoPackage>> {
+  const exclude_packages = core.getInput("exclude_packages").split(" ");
   const args = ['metadata', '--no-deps', '--format-version=1']
   const output = await captureOutput(cargoPath, args)
-  return (JSON.parse(output) as CargoMetadata).packages
+  let result = (JSON.parse(output) as CargoMetadata).packages;
+  if (exclude_packages.length > 0) {
+    result = result.filter(pack => exclude_packages.indexOf(pack.name) == -1);
```
Comment (orf) [inline src/bloat.ts:107]: Lets just use `exclude_packages.includes(pack.name)` instead of `indexOf`?

**Comment by orf** [comment]
The perms are all messed up, which is why it failed to post a message here. If you fix the above comment I'll merge this 🎉

**Comment by orf** [comment]
Can you also add `exclude_packages` to the `action.yml`?

**Comment by Vrixyz** [comment]
thanks for your responsiveness, I'm currently testing this MR on my master branch as the ci rules don't apply for regular branches 😄

File: src/bloat.ts [hunk 2]
```diff
@@ -99,7 +99,12 @@ export async function runCargoTree(cargoPath: string, packageName: string): Prom
 }
 
 export async function getCargoPackages(cargoPath: string): Promise<Array<CargoPackage>> {
+  const exclude_packages = core.getInput("exclude_packages").split(" ");
   const args = ['metadata', '--no-deps', '--format-version=1']
   const output = await captureOutput(cargoPath, args)
-  return (JSON.parse(output) as CargoMetadata).packages
+  let result = (JSON.parse(output) as CargoMetadata).packages;
+  if (exclude_packages.length > 0) {
+    result = result.filter(pack => exclude_packages.includes(pack.name) == -1);
```
Comment (orf) [inline src/bloat.ts:107]: `includes()` returns a boolean, so you can remove the `== -1` here 👍

Comment (Vrixyz) [reply src/bloat.ts]: I definitely should install [act](https://github.com/nektos/act) to avoid embarrassing myself 😉

**Comment by Vrixyz** [comment]
I think this MR is ready, here it is running correctly on my master branch: https://github.com/Vrixyz/cargo-bloat-action/runs/868841344?check_suite_focus=true

- Sorry for the commit messages not that explicit
- It's my first encounter with a github-action repository so please be wary 👀

**Comment by orf** [comment]
Thank you @Vrixyz! I'll tag and push a release later tonight 🎉

**Comment by orf** [comment]
Released v1.2.0 👍

**Pull request merged**